### ISSUE TITLE
Wrong value of wcwidth for emoji

### ISSUE THREAD

**Issue by nacknime-official** [neovim/neovim#12523]
- `nvim --version`: 
```
NVIM v0.5.0-547-gdd4018947
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-410d26af-e3fc-46f1-9ea4-44eef6b04ebd

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```
- `vim -u DEFAULTS` (version: `8.2.814`) behaves differently? `particulary yes`
- `OS`: `Archlinux`
- `WM`: `Sway (wayland)`
- `CPU`: `Intel Core i7-8550U`
- `GPU`: `Intel UHD Graphics 620 (rev 07)`
- `Driver`: `mesa 20.1.1-1`
- `Terminal / version`: `kitty 0.18.0 created by Kovid Goyal`
- `$TERM`: `xterm-256color`

### Expected behaviour
see in https://github.com/kovidgoyal/kitty/issues/2786

**Comment by teto** [comment]
for what character ? we use libc's wcwidth if you want to check

**Comment by nacknime-official** [comment]
> for what character ? we use libc's wcwidth if you want to check

maybe for all emojis. probably, I've named this issue wrong way, I'm sorry, but please, check the issue that I left in the end of this issue text

**Comment by teto** [comment]
basically terminal and nvim and fonts have to agree, that's one of the problem without good solutions at the time. You can try unsetting the 'emoji' option or read the ambiwidth option to see if that helpo. Otherwise https://github.com/thestinger/termite uses libc's wcwidth so it may work better for you.

**Comment by teto** [comment]
not sure there is a good solution here, all components need to use the same source of truth.
btw may be related https://github.com/neovim/neovim/issues/7151#issuecomment-652421539

**Comment by zeertzjq** [comment]
Either fixed by #19686 or duplicate of #7151
